src/android/CodePush.java

```diff
@@ -109,34 +96,22 @@ private boolean execIsFailedUpdate(CordovaArgs args, CallbackContext callbackCon
         return true;
     }
 
-    private boolean execApply(CordovaArgs args, CallbackContext callbackContext) {
+    private boolean execInstall(CordovaArgs args, CallbackContext callbackContext) {
         try {
             final String startLocation = args.getString(0);
             final int updateSuccessTimeoutInMillis = args.optInt(1);
+            final InstallMode installMode = InstallMode.fromValue(args.optInt(2));
 
             File startPage = this.getStartPageForPackage(startLocation);
             if (startPage != null) {
                 /* start page file exists */
                 /* navigate to the start page */
-                this.navigateToFile(startPage);
-                /* this flag will clear when reloading the plugin or after a didUpdate call from the JS layer */
-                this.didUpdate = true;
-
-                if (updateSuccessTimeoutInMillis > 0) {
-                /* start countdown for success */
-                    CodePush.ApplySucceeded = false;
-                    final CountDownTimer successTimer = new CountDownTimer(updateSuccessTimeoutInMillis, updateSuccessTimeoutInMillis) {
-                        @Override
-                        public void onTick(long millisUntilFinished) {
-                        /* empty - we are not interested in progress updates */
-                        }
-
-                        @Override
-                        public void onFinish() {
-                            onSuccessTimerFinish();
-                        }
-                    };
-                    successTimer.start();
+                if (InstallMode.IMMEDIATE.equals(installMode)) {
+                    this.navigateToFile(startPage);
```

Comment:
Should we call handleAppStart() here? By doing this, the timestamp comparisons will not happen during an immediate install right?